core/src/main/java/org/elasticsearch/gateway/MetaDataStateFormat.java

```diff
@@ -270,6 +267,8 @@ public  T loadLatestState(ESLogger logger, Path... dataLocations) throws IOExcep
                             files.add(pav);
                         }
                     }
+                } catch (NoSuchFileException | FileNotFoundException ex) {
+                    // no _state directory -- move on
```

Comment:
Are there other files that might not be in there and we're ok with that? Should we log a warning or something? I suspect its just fine for the directory not to exist but if some file inside the directory doesn't exist when the directory does thats probably bad. Like, in production. In tests is fine to just eat the exception.